### PR TITLE
Better error messages when a vector is missing from dataframe

### PR DESCRIPTION
**Pull request by sivagollapalli**
#204

### CODE REVIEW

File: spec/index/index_spec.rb [hunk 1]
```diff
@@ -362,5 +362,13 @@
...
       end
     end
+
+    context "#by_single_key" do
+      let(:df) { Daru::DataFrame.new(a: [1, 2, 3], b: [4, 5, 6] ) }
+
+      it 'raise error when vector is missing from dataframe' do
+	expect { df[:c] }.to raise_error(IndexError, /Specified vector :c does not exist/)
+      end
+    end
```
Comment (athityakumar) [inline spec/index/index_spec.rb]: @zverok - Should this test be shifted to `spec/dataframe_spec.rb`?

**Comment by v0dro** [comment]
@sivagollapalli you've made changes to index.rb and are displaying an error message only for Dataframes with missing vectors. However, an Index is used by both Vector and DataFrame and therefore returning an error message stating a vector is missing from `index.rb` is not justified.

If you want better error reporting, you should catch the IndexError in the DataFrame class and change the error message to reflect the fact that a vector is missing from a dataframe.

File: spec/index/index_spec.rb (hunk 1 above)
Comment (v0dro) [inline spec/index/index_spec.rb]: Yes this test should be in dataframe_spec.

**Comment by sivagollapalli** [comment]
@v0dro @zverok Could you check the changes?

**Comment by zverok** [comment]
Please fix Rubocop (https://travis-ci.org/SciRuby/daru/jobs/225581190, you can check it locally by running `rubocop` command) before merge.

**Pull request merged**